pkg/kubeclient/cpufrequency.go

```diff
@@ -47,6 +51,39 @@ type NodeCpuFrequencyGetter struct {
 	kubeClient      *kubernetes.Clientset
 	busyboxImage    string
 	imagePullSecret string
+	backoffFailures map[string]*backoffFailure
+}
+
+type backoffFailure struct {
+	// Number of times job failed since last success or since start
+	failedTimes float64
+	// Last failure timestamp
+	lastTimestamp time.Time
+	initialDelay  time.Duration
+}
+
+func newBackoff(delay time.Duration) *backoffFailure {
+	return &backoffFailure{initialDelay: delay}
+}
+
+// Each failure would add a 2 exponent times initialDelay to backoff
+// Any backoff call before that would return true meaning "back off"
+func (b *backoffFailure) backoff() bool {
+	return !b.lastTimestamp.IsZero() &&
+		b.lastTimestamp.Add(time.Duration(math.Pow(2,
+			b.failedTimes))*b.initialDelay).After(time.Now())
+}
+
+func (b *backoffFailure) setFailure() {
+	if b.lastTimestamp.IsZero() {
+		b.lastTimestamp = time.Now()
+	}
+	b.failedTimes++
+}
```

Comment:
Is there a limit on the max backoff time? If not, shall we set one (and maybe make it configurable)?

Reply:
There is no limit currently. The failure scenarios are rarely autocorrected, except transient network access issues, meaning updating kubeturbo deployment configuration needs update, thereby requiring restart.
If you think we should I will introduce a max delay. I however certainly don't see a use case for it to be configurable. Please let me know.

Reply:
OK. Thanks for clarifying.